src/client/io_tracker.cpp

```diff
@@ -207,6 +216,54 @@ void IOTracker::DoWrite(MDSClient* mdsclient, const FInfo_t* fileInfo) {
     }
 }
 
+void IOTracker::StartDiscard(off_t offset, size_t length, MDSClient* mdsclient,
+                             const FInfo* fileInfo,
+                             DiscardTaskManager* taskManager) {
+    offset_ = offset;
+    length_ = length;
+    type_ = OpType::DISCARD;
+
+    DoDiscard(mdsclient, fileInfo, taskManager);
+}
+
+void IOTracker::StartAioDiscard(CurveAioContext* ctx, MDSClient* mdsclient,
+                                const FInfo_t* fileInfo,
+                                DiscardTaskManager* taskManager) {
+    aioctx_ = ctx;
+    offset_ = ctx->offset;
+    length_ = ctx->length;
+    type_ = OpType::DISCARD;
+
+    DoDiscard(mdsclient, fileInfo, taskManager);
+}
+
+void IOTracker::DoDiscard(MDSClient* mdsClient, const FInfo* fileInfo,
+                          DiscardTaskManager* taskManager) {
+    LOG(INFO) << "DoDiscard offest = " << offset_ << ", length = " << length_;
+
+    int ret = Splitor::CalcDiscardSegments(this);
+
+    if (ret == 0 && !discardSegments_.empty()) {
+        for (auto index : discardSegments_) {
+            timespec abstime =
+                butil::milliseconds_from_now(discardOption_.discardTaskDelayMs);
+            if (!taskManager->ScheduleTask(index, mc_, mdsClient, abstime)) {
+                LOG(ERROR) << "Schedule discard task failed";
+                ret = -1;
+                continue;
```

Comment:
Removed this line, and always return success for discard operations.  Because return error may cause filesystem error, and failed discard operations have no bad effect.